### PR TITLE
Improve `fn new`

### PR DESCRIPTION
**Pull request by chatblanc-ciel**
change return ``Result``

test setting

### CODE REVIEW

File: arci/src/clients/partial_joint_trajectory_client.rs [hunk 1]
```diff
@@ -30,13 +30,33 @@ impl<C> PartialJointTrajectoryClient<C>
...
+            return Err(format!(
+                "Joint name length is illegal. (full = { }, partial = { })",
+                full_joint_names.len(),
+                joint_names.len()
+            ));
+        }
+
+        let mut partial = Vec::with_capacity(joint_names.len());
+        for joint_name in joint_names.iter() {
+            if let Some(_) = full_joint_names.iter().find(|&x| x == joint_name) {
```
Comment (a-miyadera) [inline arci/src/clients/partial_joint_trajectory_client.rs:45]: I think it's enough to return Err at the moment when any item of joint_names is not found in full_joint_names.

Comment (chatblanc-ciel) [reply arci/src/clients/partial_joint_trajectory_client.rs]: I changed it.
Please, check.

File: arci-urdf-viz/src/client.rs [hunk 2]
```diff
@@ -83,7 +83,8 @@ fn create_joint_trajectory_clients_inner(
             .into());
         }
         let client =
-            arci::PartialJointTrajectoryClient::new(config.joint_names, all_client.clone());
+            arci::PartialJointTrajectoryClient::new(config.joint_names, all_client.clone())
```
Comment (chatblanc-ciel) [inline arci-urdf-viz/src/client.rs:86]: Config.joint_names is 0 length.
What's happen?

Comment (OTL) [reply arci-urdf-viz/src/client.rs:86]: I think it should be an error.

File: arci/src/clients/partial_joint_trajectory_client.rs [hunk 3]
```diff
@@ -30,13 +30,59 @@ impl<C> PartialJointTrajectoryClient<C>
...
+
+        // check redundant joint name of partial
+        let input_len = joint_names.len();
+        let mut unique_joint_names = joint_names
+            .clone()
+            .into_iter()
+            .collect::<HashSet<String>>()
+            .into_iter()
+            .collect::<Vec<String>>();
+        unique_joint_names.sort();
```
Comment (a-miyadera) [inline arci/src/clients/partial_joint_trajectory_client.rs:62]: Is it necessary to sort?
And, is it necessary to be Vec? (HashSet also has `len` method.)

Comment (chatblanc-ciel) [reply arci/src/clients/partial_joint_trajectory_client.rs:62]: I forget to eliminate sort. It is Relics of the past.
Changing Vec is same.
In preview version, unique is necessary for Partial

File: ros_entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (a-miyadera) [inline ros_entrypoint.sh:1]: This file is not needed.
Please, delete.

**Review by a-miyadera — APPROVED** [review]

**Pull request merged**